modules/optimization/doc/content/modules/optimization/examples/material_transient.md

```diff
@@ -0,0 +1,99 @@
+distribution of zero will not be able to find a thermal conductivity distribution that minimizes the
+temperature misfit. To help the optimization algorithm out, we provide an initial guess for the
+spatially varying thermal conductivity. In [sensitivity],
+we plot the objective function as we vary the thermal conductivity at the
+location
+(0.75, 0.75). The slope of the objective function with respect to variations in the parameter can
+have implication in the overall convergence of the optimization problem.
+
+!media large_media/optimization/material_transient_of.png
+       style=width:40%;margin:auto;padding-top:2.5%;background-color: white;color: black;
+       id=sensitivity
+       caption=Sensitivity of the objective function when changing the diffusivity parameter at
+       the location (0.75, 0.75) around its optimal value '0.05'.
+
+
+## Problem Input File Setup
+
+This example makes use of the [TransientAndAdjoint](TransientAndAdjoint.md) executioner, which automatically computes the adjoint by transposing the forward problem's Jacobian.  This frees the user from having to derive and implement an adjoint input file based on the forward problem kernels.  This also solves the forward and adjoint problems in a single sub-application. An alternative method with two child applications corresponding to the forward and adjoint problems is given in [Multi-app executioner](materialInv_ConstK.md#sec:MultiAppExecutioner).
+
+The entire inverse optimization strategy is governed by the [OptimizationReporter](OptimizationReporter.md) object.
+The forward and adjoint problems are set up in the main application input file to be solved as a `FullSolveMultiApp` type of
+`MultiApps` problem.  Controllable parameter data is transferred to the `to_forward` sub-app using the `MultiAppReporterTransfer` objects.  Objective and gradient information is then passed back to the optimization main application using another `MultiAppReporterTransfer`.
+
+!listing examples/materialTransient/optimize_auto_adjoint.i
+
+A single input file is used to simulate the physics for the optimization problem. The core of the input
+file contains the definition of the physics that the user wants to model, as with most of MOOSE's
+simulations.
+In this case, a heat diffusion problem with Dirichlet boundary conditions is built. The forward and adjoint problems are run in two separate [nonlinear systems](NonlinearSystem.md), one for the forward problem and another one for the adjoint problem.
+These nonlinear systems are inputs to an executioner that drives the simulation of both problems, i.e. the [TransientAndAdjoint](TransientAndAdjoint.md) MOOSE object.
+
+!listing examples/materialTransient/forward_and_adjoint.i block=Problem
+
+!listing examples/materialTransient/forward_and_adjoint.i block=Variables
+
+For material inversion, the derivative of the objective function with respect to the material
+parameters needs to be computed. This quantity is problem-dependent.
+
```

Comment:
This is causing a documentation error, maybe these need to be on the same line:
[ElementOptimizationDiffusionCoefFunctionInnerProduct]ElementOptimizationDiffusionCoefFunctionInnerProduct.md)